spec.bs

```diff
@@ -2266,6 +2267,19 @@ The network revocation mechanism requires the following monkeypatches to the [[F
   (<a href="https://github.com/WICG/fenced-frame/issues/191">WICG/fenced-frame#191</a>)
 </div>
 
+<div algorithm=resolve-an-origin-patch>
+  Modify the [=resolve an origin=] algorithm to include an extra parameter: a [=URL=] |url|.
+  
+  Insert these steps immediately after step 2:
+  
+  1. Let |nonce| be <var ignore>key</var>'s [=network partition key/nonce=].
+  
+  1. If the user agent's [=network revocation exemption map=][|nonce|] [=map/exists=], but [=network
+     revocation exemption map=][|nonce|][|url|] does not [=map/exist=], then:
+     
+    1. If the user agent's [=network revocation nonce set=] [=set/contains=] |nonce|, then return.
```

Comment:
The [caller of `resolve an origin`](https://fetch.spec.whatwg.org/#ref-for-resolve-an-origin%E2%91%A1) is expecting some return value, so we can probably use "failure" since it already exists in the algorithm and the caller is set up to handle it.

```suggestion
    1. If the user agent's [=network revocation nonce set=] [=set/contains=] |nonce|, then return failure.
```